### PR TITLE
feat: setup-and-push action

### PR DESCRIPTION
**Pull request by marom-sprkl**
- Sprkl setup run at the action main part.
- Push state to CI platform at the end of the job. This part run at the post part even after failure of previous steps.

I tested this action in a workflow at the sprkl-microservices-example project: [here](https://github.com/sprkl-dev/sprkl-microservices-example/blob/feat/test-post-push/.github/workflows/setup-push-test.yml). 
[the workflow run report.](https://github.com/sprkl-dev/sprkl-microservices-example/runs/7972124646?check_suite_focus=true)
the workflow ran with failure in one of the tests. The workflow failed but the push to CI platform still happened.

### CODE REVIEW

**Review by razcoen — APPROVED** [review]
Good work!

File: setup/action.yml [hunk 1]
```diff
@@ -1,7 +1,10 @@
-name: 'sprkl: setup'
-description: 'Set up sprkl'
+name: "sprkl: setup and push"
```
Comment (erez-sprkl) [inline setup/action.yml:1]: Change to 'Sprkl personal observability for GitHub Actions"

File: setup/action.yml [hunk 2]
```diff
@@ -1,7 +1,10 @@
...
+decription: "Setup sprkl and push sprkl snapshot to CI platform at the end of the job"
 
 inputs:
+  token:
+    description: ""
```
Comment (erez-sprkl) [inline setup/action.yml:6]: Add description: "Sprkl allocated token - get your token at by installing the Sprkl GitHub Application"

File: setup/src/main.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+ */
+(async () => {
+    // get the input values from the action
+    const sprklVersion = core.getInput('version');
+    const analyze = core.getInput('analyze');
+    const setEnv = core.getInput('setenv');
+
+    // run sprkl install command
+    const installCmd = `npx @sprkl/scripts@${sprklVersion} install`;
+    await exec.exec(installCmd);
```
Comment (erez-sprkl) [inline setup/src/main.ts:16]: In some cases we know the installation may fail due to problems linking in /usr/local/bin ...
I think we should at least add $HOME/.sprkl/bin to the $PATH env variable

File: setup/action.yml [hunk 4]
```diff
@@ -1,7 +1,10 @@
...
+    required: true
   setenv:
```
Comment (erez-sprkl) [inline setup/action.yml:8]: Add description: "Instrument *ALL* node scripts by default"

**Pull request merged**